File: Data Structures & Algorithms/subsets/submission-1.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> results;
        vector<int> current;

        dfs(0, nums, current, results);

        return results;
    }

private:
    void dfs(int index, vector<int>& nums, vector<int>& current, vector<vector<int>>& results){
        int n = nums.size();
        results.push_back(current);
       
        for (int i = index; i < n; i++){
            current.push_back(nums[i]);
            dfs(i+1, nums, current, results);
            current.pop_back();
        }
    }
};

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        reach = [True] + [False] * (n)
        for i in range(n):
            if reach[i] == True:
                if i+nums[i] >= n-1:
                    return True
                else:
                    for j in range(nums[i]+1):
                        reach[i+j] = True
            
        
        return False